### ISSUE TITLE
recurring events

### ISSUE THREAD

**Issue by connorbode** [nwcell/ics.js#10]
Is it possible to define recurring events using this lib?

**Comment by asr1** [comment]
You can make a recurring event, though I'm not sure the library natively allows it.

    `	    while(start <= end)
		{
		//Now update our counter to tomorrow
	
		  var eventStart = new Date(start.setHours(eventTime.getHours(), eventTime.getMinutes()));
		  var eventEnd = new Date(start.setHours(eventTimeEnd.getHours(),eventTimeEnd.getMinutes()));
		  var newDate = start.setDate(start.getDate() + 1);
	      start = new Date(newDate);
		
		   //Convert the event to a string
		 var eventStartString = (eventStart.getMonth()+1).toString().concat("/").concat(eventStart.getDate().toString()).concat("/").concat(eventStart.getFullYear().toString()).concat(" ").concat(eventStart.getHours().toString()).concat(":").concat(eventStart.getMinutes().toString());//.concat(" PM"));
		 
		 //There is a discrepancy between indexing in months, hence the + 1
		 var eventEndString = (eventEnd.getMonth()+1).toString().concat("/").concat(eventEnd.getDate().toString()).concat("/").concat(eventEnd.getFullYear().toString()).concat(" ").concat(eventEnd.getHours().toString()).concat(":").concat(eventEnd.getMinutes().toString());//.concat(" PM"));
	

			//Check to see if it's on the right day of the week.
			 for(x = 0; x < WeekDays.length; x++)
			 {
				if(eventStart.getDay().toString() == WeekDays[x])
				{   
					cal.addEvent(name, "Class",location, new Date(eventStartString) ,new Date(eventEndString));
				}
			  }
		}
	}
	`

Basically, repeat the event every day from the first event to the last, but only add it to the calendar if the day of the week you're examining is the day of the week of you want the event to occur on.

**Comment by sferoze** [comment]
+1. I have been searching for a good javascript library to generator .ics files for an iCal feed I am creating. 

But our calendar system has reoccurring events. Some reoccurring events do not have a defined end date. 

I haven't found a ics generator library written in javascript that supports reoccurring events.

**Comment by connorbode** [comment]
@sferoze Check out [my fork](https://github.com/connorbode/ics.js).  I wrote some functionality for recurring events.  Hopefully it will satisfy your need.  

Others have submitted other functionality as well.

**Comment by sferoze** [comment]
@connorbode wow thank you so much! That is perfectly what I was looking for. Developer gold.

**Comment by nwcell** [comment]
Merged PR
